.ci/scripts/distribution/it-prepare.sh

```diff
@@ -10,4 +10,4 @@ MAVEN_PROPERTIES=(
 
 # make sure to specify the profiles used in the verify goal when running preparing to go offline, as
 # these may require some additional plugin dependencies
-mvn -B -T${MAVEN_PARALLELISM} -s ${MAVEN_SETTINGS_XML} test-compile -Pprepare-offline,parallel-tests "${MAVEN_PROPERTIES[@]}"
+mvn -B -T${MAVEN_PARALLELISM} -s ${MAVEN_SETTINGS_XML} package -Pprepare-offline,parallel-tests "${MAVEN_PROPERTIES[@]}"
```

Comment:
❓ Isn't it already packaged by the build-java script? However, that one does not compile test code, so this was compiling the test code to make sure everything is available to run the tests offline in the next step. Is the test code still compiled if we just say package here? (probably yes, because we don't `skipTests` here)

Reply:
Not quite. The shared stash is actually just the .m2 repository, but there is no `target` folders in the agent, so when building the package is not available "locally". Does that make sense? So anything which tries to get the target JAR, processed resources, etc., from the local project folder will fail. We could consider stashing the target folders as well, which would solve this. 

We can also force the test-compile phase as well during the initial build by the way, but I don't know if we gain much by doing this :thinking:

Reply:
I dislike building the test code in the build step, but I don't have a good reason for it. Just a feeling. I'm totally fine with the current approach. Thanks for the explanation.